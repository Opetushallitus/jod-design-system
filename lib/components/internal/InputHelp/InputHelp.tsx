interface InputFieldProps {
  /** The help text to display below the input field */
  helpText?: string;
  /** The id for the help text element */
  id: string;
  /** Test id for querying in tests */
  dataTestId?: string;
}

export const InputHelp = ({ helpText, id, dataTestId }: InputFieldProps) => {
  return helpText !== undefined ? (
    <div
      id={id}
      className="ds:mt-2 ds:block ds:text-help ds:text-secondary-gray ds:font-arial"
      data-testid={dataTestId}
    >
      {helpText}
    </div>
  ) : null;
};
