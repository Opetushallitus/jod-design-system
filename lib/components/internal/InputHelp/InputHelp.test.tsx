import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { InputHelp } from './InputHelp';

describe('InputHelp', () => {
  const helpText = 'This is some helpful text';
  const id = 'help-text';

  it('renders help text correctly', () => {
    const { getByText, container } = render(<InputHelp helpText={helpText} id={id} />);
    expect(getByText(helpText)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('does not render when helpText is undefined', () => {
    const { container } = render(<InputHelp id={id} />);
    expect(container).toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });

  it('applies the provided id to the help text element', () => {
    const { getByText, container } = render(<InputHelp helpText={helpText} id={id} />);
    expect(getByText(helpText)).toHaveAttribute('id', id);
    expect(container).toMatchSnapshot();
  });

  it('applies the provided dataTestId to the help text element', () => {
    const dataTestId = 'custom-help-text';
    const { getByTestId, container } = render(<InputHelp helpText={helpText} id={id} dataTestId={dataTestId} />);
    expect(getByTestId(dataTestId)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
