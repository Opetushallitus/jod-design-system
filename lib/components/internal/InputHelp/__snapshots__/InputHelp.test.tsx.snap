// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputHelp > applies the provided dataTestId to the help text element 1`] = `
<div>
  <div
    class="ds:mt-2 ds:block ds:text-help ds:text-secondary-gray ds:font-arial"
    data-testid="custom-help-text"
    id="help-text"
  >
    This is some helpful text
  </div>
</div>
`;

exports[`InputHelp > applies the provided id to the help text element 1`] = `
<div>
  <div
    class="ds:mt-2 ds:block ds:text-help ds:text-secondary-gray ds:font-arial"
    id="help-text"
  >
    This is some helpful text
  </div>
</div>
`;

exports[`InputHelp > does not render when helpText is undefined 1`] = `<div />`;

exports[`InputHelp > renders help text correctly 1`] = `
<div>
  <div
    class="ds:mt-2 ds:block ds:text-help ds:text-secondary-gray ds:font-arial"
    id="help-text"
  >
    This is some helpful text
  </div>
</div>
`;
